pkg/lookup/lookup.go

```diff
@@ -82,6 +83,29 @@ func GetTemplates(
 	return templates
 }
 
+// GetMandatoryTemplates returns all ModuleTemplates for modules which are marked as mandatory with the given channel.
+func GetMandatoryTemplates(ctx context.Context, kymaClient client.Reader) (ModuleTemplatesByModuleName,
```

Comment:
I don't think it's a good idea put this function inside this package. The main purpose for the original file is designed to deal with regular template, and we have complicated lookup strategy, that's why `GetTemplates` have so many private methods depends on, but your `GetMandatoryTemplates` basically depends on nothing within this file, which decreased the code cohesion.

Try to create a new file to put this code, and you can also think of renaming `GetTemplates` to something more specific like `GetNormalTemplates` to distinguish between them.